### DISCUSSION TITLE
How to add prefix to slug?

### DISCUSSION THREAD

**Discussion by chimit** [laravel/nova-issues#4638] [Q&A]
I need to extend slug generation logic by adding a prefix to slugs. But it still should be possible to override. I tried to change the resolution logic but it doesn't work as expected. It just puts `prefix-` into the field as an initial value and is overridden on Name change:

```php
Text::make('Name'),

Slug::make('Slug', 'slug', function ($slug) {
    return 'prefix-' . $slug;
})
    ->from('Name'),
```

**Comment by anditsung** [comment]
you can add it 
- using model observer,
- using setSlugAttriubte

**Comment by chimit** [comment]
`setSlugAttriubte` won't show correct slugs dynamically in UI. I believe a model observer has the same problem.
